packages/frontend/src/page/document_menu.tsx

```diff
@@ -80,41 +89,68 @@ export function DocumentMenu(props: {
+                                </MenuItem>
+                            </Match>
+                            <Match when={docType() === "diagram"}>
+                                <MenuItem onSelect={() => onNewDiagram()}>
+                                    <DocumentTypeIcon documentType="diagram" />
+                                    <MenuItemLabel>{"New diagram"}</MenuItemLabel>
+                                </MenuItem>
+                            </Match>
+                        </Switch>
+                        <Show when={props.liveDoc.doc.type !== "analysis"}>
+                            <MenuItem onSelect={() => onNewAnalysis()}>
+                                <DocumentTypeIcon documentType="analysis" />
+                                <MenuItemLabel>{`New analysis of this ${docType()}`}</MenuItemLabel>
                             </MenuItem>
-                        </Match>
-                        <Match when={docType() === "diagram"}>
-                            <MenuItem onSelect={() => onNewDiagram()}>
-                                <DocumentTypeIcon documentType="diagram" />
-                                <MenuItemLabel>{"New diagram"}</MenuItemLabel>
-                            </MenuItem>
-                        </Match>
-                    </Switch>
-                    <Show when={props.liveDoc.doc.type !== "analysis"}>
-                        <MenuItem onSelect={() => onNewAnalysis()}>
-                            <DocumentTypeIcon documentType="analysis" />
-                            <MenuItemLabel>{`New analysis of this ${docType()}`}</MenuItemLabel>
-                        </MenuItem>
-                    </Show>
-                    <Show when={showSeparator()}>
+                        </Show>
+                        <Show when={showSeparator()}>
+                            <MenuSeparator />
+                        </Show>
+                        <DuplicateMenuItem doc={props.liveDoc.doc} />
+                        <ExportJSONMenuItem doc={props.liveDoc.doc} />
+                        <CopyJSONMenuItem doc={props.liveDoc.doc} />
                         <MenuSeparator />
-                    </Show>
-                    <DuplicateMenuItem doc={props.liveDoc.doc} />
-                    <ExportJSONMenuItem doc={props.liveDoc.doc} />
-                    <CopyJSONMenuItem doc={props.liveDoc.doc} />
-                </DropdownMenu.Content>
-            </DropdownMenu.Portal>
-        </DropdownMenu>
+                        <MenuItem
+                            disabled={!canDelete()}
+                            onSelect={async () => {
+                                // Explicitly closing the menu avoids some
+                                // strange conflict between kobalte and corvu.
+                                // Our UI locks if we don't close the menu
+                                // _first_.
```

Comment:
Ah, I've seen something like this before but I can't recall the exact circumstances. It's unfortunate.